### PR TITLE
sci-mathematics/freefem++: upgrading to freefem++-4.6

### PR DESCRIPTION
**Pull request by Fougeroc**
Adding support for latest version of FreeFem++.

### CODE REVIEW

File: sci-mathematics/freefem++/freefem++-4.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
Comment (thesamesam) [inline sci-mathematics/freefem++/freefem++-4.6.ebuild:4]: EAPI 7 if you can?

File: sci-mathematics/freefem++/freefem++-4.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+DESCRIPTION="Solve PDEs using FEM on 2d and 3d domains"
+HOMEPAGE="http://www.freefem.org/ff++/"
+EGIT_REPO_URI="https://github.com/FreeFem/FreeFem-sources"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="doc examples mpi opengl X"
+
+RDEPEND="
```
Comment (thesamesam) [inline sci-mathematics/freefem++/freefem++-4.6.ebuild:17]: Sort these? (conditionals last)

File: sci-mathematics/freefem++/freefem++-4.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	X? (
+		media-fonts/font-misc-misc
+		x11-libs/libX11
+		x11-libs/libXext
+		x11-libs/libXpm
+		x11-libs/libXxf86vm
+		)"
+
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
```
Comment (thesamesam) [inline sci-mathematics/freefem++/freefem++-4.6.ebuild:38]: all of these are likely BDEPEND on EAPI 7

File: sci-mathematics/freefem++/freefem++-4.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2015 Gentoo Foundation
```
Comment (thesamesam) [inline sci-mathematics/freefem++/freefem++-4.6.ebuild:1]: Wrong header. Use repoman commit?

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: sci-mathematics/freefem++/freefem++-4.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
+
+inherit git-r3 autotools eutils flag-o-matic mpi versionator toolchain-funcs
```
Comment (epsilon-0) [inline sci-mathematics/freefem++/freefem++-4.6.ebuild:6]: It is a versioned ebuild, do not use `git-r3`

File: sci-mathematics/freefem++/freefem++-4.6.ebuild [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+DESCRIPTION="Solve PDEs using FEM on 2d and 3d domains"
+HOMEPAGE="http://www.freefem.org/ff++/"
+EGIT_REPO_URI="https://github.com/FreeFem/FreeFem-sources"
```
Comment (epsilon-0) [inline sci-mathematics/freefem++/freefem++-4.6.ebuild:10]: Again, same as above, its versioned so use a `SRC_URI`

File: sci-mathematics/freefem++/freefem++-4.6.ebuild [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	doc? (
+		dev-texlive/texlive-latexrecommended
+		dev-texlive/texlive-latexextra
+		virtual/latex-base
+		media-gfx/imagemagick
+		)"
+
+src_prepare() {
```
Comment (epsilon-0) [inline sci-mathematics/freefem++/freefem++-4.6.ebuild:46]: No need for this, it is done automatically.

**Review by epsilon-0 — CHANGES_REQUESTED** [review]

**Comment by AndrewAmmerlaan** [comment]
Closing as the latest version of freefem++ in ::sci is 4.7.1 already.

**Pull request closed without merging**